crates/matrix-sdk/src/sliding_sync/cache.rs

```diff
@@ -167,10 +166,7 @@ pub(super) async fn restore_sliding_sync_state(
             trace!("Successfully read the `SlidingSync` from the cache");
             // Let's update the `SlidingSync`.
             if let Some(since) = to_device_since {
-                let to_device_ext = &mut extensions.get_or_insert_with(Default::default).to_device;
-                if to_device_ext.enabled == Some(true) {
-                    to_device_ext.since = Some(since);
-                }
+                *to_device_token.get_or_insert_with(Default::default) = since;
```

Comment:
Why not simply:

```rust
*to_device_token = Some(since);
```

?

Reply:
Yes indeedy, I got confused by the previous code, cheers.